### PR TITLE
Use lists instead of hashsets for injectable caching

### PR DESCRIPTION
**Pull request by firtina-improbable**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
Removal of some nasty array magic from the hot loop
#### Tests
Tested with local deployment, monobehaviour examples and world commands still work
#### Documentation
Very internal
#### Primary reviewers
If your change will take a long time to review, you can name at most two primary reviewers who are ultimately responsible for reviewing this request. @ mention them.

@samiwh @zeroZshadow

### CODE REVIEW

File: workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/GameObjectWorldCommandSystem.cs [hunk 1]
```diff
@@ -65,8 +64,9 @@ protected override void OnUpdate()
 
                 foreach (var receivedResponse in reserveEntityIdsResponseData.CommandResponses[i].Responses)
                 {
-                    foreach (var worldCommandHandler in worldCommandResponseHandlers)
+                    foreach (var injectable in worldCommandResponseHandlers)
                     {
+                        var worldCommandHandler = (WorldCommands.Requirables.WorldCommandResponseHandler) injectable;
```
Comment (samiwh) [inline workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/GameObjectWorldCommandSystem.cs]: There is a trick we can do to remove these casts. `IEnumerable` is `covariant`, so internally we can store these as `IEnumerable<IInjectable>` then have the `TryGet<>` function cast it back to `List<Concrete`>.

Comment (samiwh) [reply workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/GameObjectWorldCommandSystem.cs]: Covariance means you can cast the `IEnumerable<IInjectable>` to the concrete list directly. If we are iterating over the erased types we can cast directly to `List<IInjectable>`.

Comment (samiwh) [reply workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/GameObjectWorldCommandSystem.cs]: This isn't too far removed from just casting it to object though at that point. Although might be slightly more readable.

Comment (samiwh) [reply workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/GameObjectWorldCommandSystem.cs]: Arguably it's dangerous to do things that allow you to cast to the wrong type but we have to do the cast somewhere anyway so I don't think we are actually risking anything.

File: workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/GameObjectWorldCommandSystem.cs [hunk 2]
```diff
@@ -132,8 +134,7 @@ protected override void OnUpdate()
                 return null;
             }
 
-            return Array.ConvertAll(injectables.ToArray(),
```
Comment (jared-improbable) [inline workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/GameObjectWorldCommandSystem.cs]: Owwwww...

Comment (firtina-improbable) [reply workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/GameObjectWorldCommandSystem.cs]: Yes I lost a few DKPs over this

**Comment by MPeti01** [comment]
Can confirm no duplicates.

File: workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/GameObjectWorldCommandSystem.cs (hunk 1 above)
Comment (MPeti01) [inline workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/GameObjectWorldCommandSystem.cs]: Could make it read nicer by casting inside the loop: `for (WorldCommands.Requirables.WorldCommandResponseHandler worldCommandHandler in worldCommandResponseHandlers)`

Comment (MPeti01) [reply workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/GameObjectWorldCommandSystem.cs]: Also twice below

**Comment by MPeti01** [comment]
I assume this was held for the milestone feature freeze, could it be resumed now?

**Comment by firtina-improbable** [comment]
I think so, cc @callumlawson (for when he's back from vacation)

**Comment by firtina-improbable** [comment]
@callumlawson thoughts?

**Comment by jessicafalk** [comment]
What is happening with this? @firtina-improbable @callumlawson

**Comment by firtina-improbable** [comment]
I'll rebase it

**Comment by firtina-improbable** [comment]
Rebased.

**Comment by firtina-improbable** [comment]
confirming world commands and other monobehaviour bits still work

**Pull request merged**